addon/components/au-date-picker.js

```diff
@@ -1,72 +1,66 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
-import { assert, runInDebug } from '@ember/debug';
+import { assert, deprecate, runInDebug } from '@ember/debug';
 import { guidFor } from '@ember/object/internals';
+import { getOwnConfig, macroCondition } from '@embroider/macros';
 import formatISO from 'date-fns/formatISO';
 
 const DEFAULT_ADAPTER = {
   parse: function (value = '', createDate) {
-    const matches = value.match(/^(\d{1,2})-(\d{1,2})-(\d{4})$/);
+    let dateRegex = macroCondition(getOwnConfig().dutchDatePickerLocalization)
+      ? /^(\d{1,2})\.(\d{1,2})\.(\d{4})$/
+      : /^(\d{1,2})-(\d{1,2})-(\d{4})$/;
+
+    const matches = value.match(dateRegex);
     if (matches) {
       return createDate(matches[3], matches[2], matches[1]);
     }
   },
   format: function (date) {
-    return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
+    let separator = macroCondition(getOwnConfig().dutchDatePickerLocalization)
+      ? '.'
+      : '-';
+    let day = `${date.getDate()}`.padStart(2, '0');
+    let month = `${date.getMonth() + 1}`.padStart(2, '0');
+    return `${day}${separator}${month}${separator}${date.getFullYear()}`;
   },
 };
 
 const DEFAULT_LOCALIZATION = {
-  buttonLabel: 'Choose date',
-  placeholder: 'DD-MM-YYYY',
-  selectedDateMessage: 'Selected date is',
-  prevMonthLabel: 'Previous month',
-  nextMonthLabel: 'Next month',
-  monthSelectLabel: 'Month',
-  yearSelectLabel: 'Year',
-  closeLabel: 'Close window',
-  keyboardInstruction: 'You can use arrow keys to navigate dates',
-  calendarHeading: 'Choose a date',
-  dayNames: [
-    'Sunday',
-    'Monday',
-    'Tuesday',
-    'Wednesday',
-    'Thursday',
-    'Friday',
-    'Saturday',
-  ],
-  monthNames: [
-    'January',
-    'February',
-    'March',
-    'April',
-    'May',
-    'June',
-    'July',
-    'August',
-    'September',
-    'October',
-    'November',
-    'December',
-  ],
-  monthNamesShort: [
-    'Jan',
-    'Feb',
-    'Mar',
-    'Apr',
-    'May',
-    'Jun',
-    'Jul',
-    'Aug',
-    'Sep',
-    'Oct',
-    'Nov',
-    'Dec',
-  ],
+  dayNames: getLocalizedDays(),
+  monthNames: getLocalizedMonths(),
+  monthNamesShort: getLocalizedMonths('short'),
 };
 
+if (macroCondition(getOwnConfig().dutchDatePickerLocalization)) {
+  Object.assign(DEFAULT_LOCALIZATION, {
+    buttonLabel: 'Kies een datum',
+    placeholder: 'DD.MM.JJJJ',
```

Comment:
I've used `DD.MM.JJJJ` for now since it seems the most official, but it can easily be updated when ABB decides on the format we should use across projects.